pkg/controllers/cstorvolumeconfig/deployment.go

```diff
@@ -384,7 +386,7 @@ func (c *CVCController) BuildTargetDeployment(
 		WithSelectorMatchLabelsNew(getDeployMatchLabels(vol.Name)).
 		WithPodTemplateSpec(
 			apicore.NewPodTemplateSpec().
-				WithLabelsNew(getDeployTemplateLabels(vol.Name)).
+				WithLabelsNew(getDeployTemplateLabels(vol.Name, vol.GetLabels()[openebsPVC])).
```

Comment:
In [PR](https://github.com/openebs/cstor-csi/pull/100/files) you added it under annotation but here we are getting from labels

Reply:
Got to see vol is cStorVolume resource